### ISSUE TITLE
Saved wallets

### ISSUE THREAD

**Issue by leafcutterant** [MyCryptoHQ/MyCrypto#2288]
### Description of the Feature

tl;dr: An option to save/reopen wallets that need the entering of private data.

I talk about these types:
* private key
* mnemonic phrase

These are marked as insecure, but it's important to keep in mind that most people don't (and potentially won't) have a hardware wallet and some avoid them on purpose.

Using MyCrypto with these is not only cumbersome (having to store the data somewhere else and enter it every time), it is also dangerous because the private data may be intercepted while on the clipboard, or worse, if the user is inexperienced, they may develop a habit of storing private data in plaintext.

To this end, I propose the option to save these wallets in MyCrypto and offer to open them after the user logged out/exited the app.

The option to save the privkey/mnemonic would be offered as a button once the user entered the respective data and pressed the "unlock"/"choose address" button. This would open a new pop-up flow. In order to save a wallet, MyCrypto would require the user to name the wallet and define a password (plus enter it again to rule out typos). All the wallet's content would be encrypted with the password, the only thing accessible without the password would be the wallet's name. Once done, MyCrypto would save the encrypted wallet in the application directory, and the user could go on using the wallet or log out.

Once the "View & Send" screen is accessed again, MyCrypto would display the saved wallets either under a new section ("Saved wallets") or within the wallet type they correspond to (e.g. a drop-down list of saved wallets of a type would appear once you choose that wallet type). For the sake of privacy, as previously described, only the wallet name would be accessible and displayed before the wallet is successfully unlocked. Choosing a saved wallet would prompt the user for the password. Upon successfully unlocking the wallet, the UX would proceed as usual.

For the sake of privacy and reversibility, MyCrypto would allow the deletion of these saved wallets. This could be implemented in the form of a delete button next to the saved wallet's name in the list. A deletion would require entering the saved wallet's password. If someone gets stuck with a saved wallet which they no longer want to use but forgot the password of, it could be made possible to delete the wallet via some manual mucking-about in the application directory, or with ticking off an option upon uninstalling MyCrypto.

### Example(s) of Feature

None yet in Ethereum -- so it's a great first-mover opportunity!

What do you think?

**Comment by tayvano** [comment]
This is partially addressed by GAU—especially possible to expand upon current plans for GAU in desktop application. Good insights here.

**Comment by leafcutterant** [comment]
Sorry, could you point me to what GAU is?

**Comment by tayvano** [comment]
It's the new stuff we are working on right now that basically improves the global experience across the app and allows viewing multiple accounts simultaneously without having to unlock their individual accounts each time they want to view information, etc. A lot of what you are talking about but slightly different. I think it will answer a lot of your requests and probably inspire a bunch more. 😉

**Comment by leafcutterant** [comment]
Cool! Looking forward to it :)

**Comment by leafcutterant** [comment]
@tayvano Where can I read more about GAU?

**Comment by wtzb** [comment]
Thank you for opening this issue! While this is still present on our current site, the work we're doing on the new version of MyCrypto should make it obsolete. Please come check out the beta [here](https://beta.mycrypto.com/).

Thanks for helping us and letting us know of any ideas you might have, as it really helps us improve our product. I'm closing this issue here, but please know that we really appreciate any feedback you might have :)
